packages/react-dialog/Spec.md

```diff
@@ -2,62 +2,302 @@
+    <div class="fui-dialog-body">Do you want to send this message without a subject?</div>
+    <footer class="fui-dialog-footer">
+      <button>Send</button>
+      <button>Cancel<button>
+    </footer>
+  </div>
+</div>
+```
 
 ## Migration
 
-_Describe what will need to be done to upgrade from the existing implementations:_
+_TBA: Link to migration guide doc_
+
+## Behaviours
+
+### Mouse & touch
+
+#### Modal
+
+![Mouse and touch behaviour of Modal dialog](./assets/modal-mouse-touch.png)
+
+1. Clicking on the trigger (element / button component) a Dialog is displayed with a dimmed background.
+2. Windows under the dialog are inert.
+3. Clicking on the dimmed background dismisses the dialog.
+4. Clicking the dismiss button (X icon in header), or cancel/dismiss button (footer) will close the dialog.
+5. The dialog can be dismissed also when confirmation button is clicked (footer).
+
+#### Non-modal
+
+![Mouse and touch behaviour of a non-modal dialog](./assets/non-modal-mouse-touch.png)
+
+1. Clicking on the trigger (element / button component) a Dialog is displayed without a dimmed background
+2. A user can continue to interact with elements on the page behind the dialog
+3. Clicking the dismiss button (X icon in header), or cancel/dismiss buttons (footer) will close the dialog
+
+#### Alert dialog
+
+![Mouse and touch behaviour of alert dialog](./assets/alert-mouse-touch.png)
 
-- _Migration from v8_
-- _Migration from v0_
+1. Clicking on the trigger (element / button component) a Dialog is displayed with a dimmed background
+2. Widnows under the dialog are are inert and clicking on the dimmed background will not close the dialog
+3. Clicking the dismiss button (X icon in header), or cancel/dismiss buttons (footer) will dismiss the dialog
 
-## Behaviors
+### Keyboard
 
-_Explain how the component will behave in use, including:_
+#### Modal
 
-- _Component States_
-- _Interaction_
-  - _Keyboard_
-  - _Cursor_
-  - _Touch_
-  - _Screen readers_
+![Keyboard behaviour of a modal dialog](./assets/modal-keyboard.png)
+
+1. **(1)** TabKey to set focus on Trigger, use EnterKey to open.
+2. **(2-5)** Focus is moved to the first focusable control inside the dialog.
+3. **(5-6)** After the dialog is dismissed, keyboard focus should be moved back to where it was before it moved into the dialog. Otherwise the focus can be dropped to the beginning of the page. Or if the item is no longer available it can be moved to the next logical location in that region i.e. next / previous item.
+4. **TabKey** Moves focus to next focusable element inside the dialog. When focus is on the last focusable element in the dialog, moves focus to the first focusable element in the dialog.
+5. **Shift+Tab** Moves focus to previous focusable element inside the dialog. When focus is on the first focusable element in the dialog, moves focus to the last focusable element in the dialog.
+6. **EscKey** Closes the dialog.
```

Comment:
are we gonna provide a functionality to be able to turn this off ? (not sure if that's good idea heh)

Reply:
No I don't think we should give an option to turn off this functionality because it might make the dialogs unusable to keyboard users. This approach follows the WCAG guidelines.